squad_3_ad_data_science/main.py

```diff
@@ -67,6 +146,78 @@ def metadata(**kwargs):
     """
     print("==> TESTING MODEL PERFORMANCE AND GENERATING METADATA")
 
+    # Kwargs processing
+    if 'update' in kwargs.keys():
+        update = kwargs['update']
+        logger.info(f'Update flag found.')
+
+    else:
+        update = True
+
+    # Load saved labeled data
+    labeld_data_pkl_path = join(config.data_path,
+                                config.TRAIN_DATA_PKL_LABELED)
+    data = pd.read_pickle(labeld_data_pkl_path)
+
+    # TODO apply Nathan's function to calculate Average PRecision for each csv on tests
+
+    # Try to load previous test performance file, to add new tests
+    if isfile(config.performance_metadata_path):
+        logger.info(f'File `{config.performance_metadata_path}` ' +
+                    'found. Reading...')
+        with open(config.performance_metadata_path) as jf:
+            perf = json.load(jf)
+
+    else:
+        logger.info(f'No previous perfomance file found. Creatings new...')
+        perf = dict()
+
+    # Run over each test file, loading its ids and running the prediction
+    for f in config.test_datasets:
+        if not isfile(f):
+            logger.error(f'Test file `{f}` not find. Ignoring...`')
+            continue
+
+        logger.info(f'Processing file `{f}`...')
+
+        # Label to identify register
+        name = f.split('/')[-1][:-4]
```

Comment:
What are you trying to get here?

Reply:
Looping over three test files (`estaticos_market<1,2,3>.csv`) to generate performance metadata with our validation methods (currently just Average Precision)

Bad idea?

Reply:
Or are you talking about line 184? There I'm trying to get a label to identify the performance record, removing `.csv` from the end of filename

Reply:
os.path lib have these functions ready. Now that you commented your intention ( you could have commented it right above the line). For instance:
````pyhton
>>> import os
>>> base=os.path.basename('/root/dir/sub/file.ext')
>>> base
'file.ext'
>>> os.path.splitext(base)
('file', '.ext')
>>> os.path.splitext(base)[0]
'file'
````

source: https://stackoverflow.com/a/678266/7299191